app/src/main/java/com/framgia/kotlintraining/moviedb/base/BaseFragment.kt

```diff
@@ -0,0 +1,51 @@
+package com.framgia.kotlintraining.moviedb.base
+
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.databinding.DataBindingUtil
+import androidx.databinding.ViewDataBinding
+import androidx.fragment.app.Fragment
+
+abstract class BaseFragment<ViewBinding : ViewDataBinding, ViewModel : BaseViewModel> : Fragment() {
+
+    lateinit var viewModel: ViewModel
+
+    abstract val layoutRes: Int
+
+    abstract fun initComponent(viewBinding: ViewBinding)
+
+    abstract fun getLayoutResource(): Int
+
+    internal fun replaceFragment(
+        fragment: Fragment,
+        container: Int,
+        TAG: String?,
+        addToBackStack: Boolean? = false,
+        transit: Int? = -1
```

Comment:
Đây cũng nên để là not null vì cậu khai báo thế này thì nó không bao giờ null được